engine/src/vault/view.rs

```diff
@@ -400,6 +537,46 @@ impl Record {
         Ok(())
     }
 
+    /// Update the key and id of an existing [`Record`].
```

Comment:
This looks fine but I do wonder if there is some way that we could introduce some obfuscation/entropy in here.  In essence, this is a read function for the record; where you decrypt a record (read it) and then re-encrypt it.   Obviously, if we are going to rekey the data, there really isn't a easy way around revealing the record even if its only for a couple of microseconds.  I almost wonder if we should explore something like proxy re-encryption to avoid exposing the secret at all.

Reply:
Definitely sounds worth exploring. I would still already merge this into `dev-refactor` so that it can be included in the security audit starting today, But I will follow up on that idea and look into proxy re-encryption and how we can use it here, before it is merged to `dev`.